pcdsdevices/interface.py

```diff
@@ -890,3 +890,49 @@ def update(self, *args, name=None, current=None, **kwargs):
                            current=current, **kwargs)
         else:
             super().update(*args, name=name, current=current, **kwargs)
+
+
+class LightpathMixin(OphydObject):
+    """
+    Mix-in class that makes it easier to establish a lightpath interface.
+
+    Use this on classes that are not state positioners but would still like to
+    be used as a top-level device in lightpath.
+    """
+    SUB_STATE = 'state'
+    _default_sub = SUB_STATE
+
+    # Component names whose values are relevant for inserted/removed
+    lightpath_cpts = []
+
+    def __init_subclass__(cls, **kwargs):
+        # Magic to subscribe to the list of components
+        super().__init_subclass__(**kwargs)
+        if not cls.lightpath_cpts:
+            raise NotImplementedError('Did not implement LightpathMixin')
+        for cpt_name in cls.lightpath_cpts:
+            cpt = getattr(cls, cpt_name)
```

Comment:
Should there be a check that the class has the lightpath interface? (inserted, removed properties)

Reply:
I don't think this is a requirement. If the class we subscribe to here isn't a lightpath-compatible object we can override `_set_lightpath_states`. I also think I should separate out the default state setter into a subclass here, since it isn't actually generic behavior.